model/infinispan/src/main/java/org/keycloak/models/sessions/infinispan/RootAuthenticationSessionAdapter.java

```diff
@@ -109,14 +115,28 @@ public AuthenticationSessionModel getAuthenticationSession(ClientModel client, S
 
     @Override
     public AuthenticationSessionModel createAuthenticationSession(ClientModel client) {
+        Map<String, AuthenticationSessionEntity> authenticationSessions = entity.getAuthenticationSessions();
+        if (authenticationSessions.size() >= authSessionsLimit) {
+            Map.Entry<String, AuthenticationSessionEntity> authSession = authenticationSessions.entrySet().stream()
+                    .sorted(Comparator.comparingInt(e -> e.getValue().getTimestamp())).findFirst().get();
```

Comment:
Some thoughts:
* It seems this code performs more sorting than necessary. Using `Collections.min` may be more efficient.
* Should we enforce that `authSessionsLimit > 0`?
* Since only the key is used below it may be nicer to call `.getKey()` here rather than twice below.
* Could `Map` be modified concurrently (sorry for asking; didn't check out the code)? If so: in case of small `authSessionsLimit` and heavy concurrency the `.get()` may fail with `NoSuchElementException`. (Yeah, _super_ unlikely, but still.)

Reply:
Something I forgot to mention: since `Comparator.comparingInt(e -> e.getValue().getTimestamp())` is invariant it could be moved to a `private static final` constant.

Reply:
Thank you, @Stephan202, good spot!

> Some thoughts:
> 
> * It seems this code performs more sorting than necessary. Using `Collections.min` may be more efficient.

You are right, `Stream.min` could be actually preferred.

> * Should we enforce that `authSessionsLimit > 0`?

+1

> * Since only the key is used below it may be nicer to call `.getKey()` here rather than twice below.

+1

> * Could `Map` be modified concurrently (sorry for asking; didn't check out the code)? If so: in case of small `authSessionsLimit` and heavy concurrency the `.get()` may fail with `NoSuchElementException`. (Yeah, _super_ unlikely, but still.)

This one should not, so this is not a concern for this PR.

Reply:
@Stephan202 thank you! I included your ideas.